docs/Flatten-transform.md

```diff
@@ -0,0 +1,40 @@
+# Flatten Transform
+
+## Description
+Flatten is a transform plugin that flattens nested data structures.
+
+## Use Case
+
+The transform is used  to convert a nested data structure  into a single flattened record where each key in the record 
+is a ```_``` name path to the node in the nested structure. 
+
+Configuration
+-------------
+**Fields to flatten:** Specifies the list of fields in the input schema to be flattened.
+
+**Level to limit flattening:** Limit flattening to a certain level in nested structures.
+
+**Prefix:** An optional prefix to be used while generating names of the flattened fields. This can be used to fix conflicts that can occur if fields have the same name after flattening. The prefix is added as <prefix>_<flattened_name >.
```

Comment:
This will not fix all the conflicts. I dont think prefix is needed. @bdmogal could you please confirm that.
When there is a conflict:
1. You get a validation error (not runtime)
2. The error has a clear call to action. "Rename x, y, or z fields before using the flatten transform"